node/src/store.rs

```diff
@@ -1,9 +1,214 @@
+use celestia_types::ExtendedHeader;
+use dashmap::mapref::entry::Entry;
+use dashmap::DashMap;
+use std::sync::atomic::{AtomicU64, Ordering};
+use tendermint::Hash;
+use thiserror::Error;
+use tracing::{error, info, instrument};
+
+trait StoreTrait {
```

Comment:
please rename it to `Store` and then add a meaningful name for implementation, like eg. `InMemoryStore`. We will have a few stores and likely will refer to them by the trait name in most places

Reply:
renamed :+1: